dev/archery/archery/release/core.py

```diff
@@ -361,6 +362,40 @@ def commits(self):
         commit_range = f"{lower}..{upper}"
         return list(map(Commit, self.repo.iter_commits(commit_range)))
 
+    @cached_property
+    def default_branch(self):
+        default_branch_name = os.getenv("DEFAULT_BRANCH")
+
+        if default_branch_name is None:
+            try:
+                # Set up repo object
+                arrow = ArrowSources.find()
+                repo = Repo(arrow.path)
+                origin = repo.remotes["origin"]
+                origin_refs = origin.refs
+
+                # Get git.RemoteReference object to origin/HEAD
+                origin_head = origin_refs["HEAD"]
+
+                # Get git.RemoteReference object to origin/main or
+                # origin/master
+                origin_head_reference = origin_head.reference
+
+                # Get string value of remote head reference, should return
+                # "origin/main" or "origin/master"
+                origin_head_name = origin_head_reference.name
+                origin_head_name_tokenized = origin_head_name.split("/")
+
+                # The last token is the default branch name
+                default_branch_name = origin_head_name_tokenized[-1]
```

Comment:
This is roughly the same logic in as `dev/archery/archery/crossbow/core.py`, right? 
Perhaps factor it out in the `Repository` class?

Reply:
Although the code in both classes perform the same functions, they are not using the same Git libraries. While `dev/arhcery/archery/crossbow/core.py` uses `pygit2`, `dev/archery/archery/release/core.py` uses `GitPython`. The `Repo` class that is used in each module are also not shared.

In an effort to limit dependencies, I wrote the code to work with the libraries that were already being imported.

Reply:
I've created a JIRA ticket to refactor archery to not require 4 different git related dependencies: https://issues.apache.org/jira/browse/ARROW-18038

Reply:
Thanks for creating this Jira ticket